Simplenote/src/main/java/com/automattic/simplenote/CustomAppLock.java

```diff
@@ -0,0 +1,233 @@
+package com.automattic.simplenote;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.text.TextUtils;
+import android.util.Base64;
+
+import org.wordpress.passcodelock.AbstractAppLock;
+import org.wordpress.passcodelock.BuildConfig;
+import org.wordpress.passcodelock.PasscodeUnlockActivity;
+import org.wordpress.passcodelock.StringUtils;
+
+import java.util.Date;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.DESKeySpec;
+
+public class CustomAppLock extends AbstractAppLock {
+    public static boolean isSupportedApi() {
+        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH;
+    }
+
+    private static final String UNLOCK_CLASS_NAME = PasscodeUnlockActivity.class.getName();
+    private static final String OLD_PASSWORD_SALT = "sadasauidhsuyeuihdahdiauhs";
+    private static final String OLD_APP_LOCK_PASSWORD_PREF_KEY = "wp_app_lock_password_key";
+
+    private Application mCurrentApp;
+    private SharedPreferences mSharedPreferences;
+    private Date mLostFocusDate;
+
+    public CustomAppLock(Application app) {
+        super();
+        mCurrentApp = app;
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(mCurrentApp);
+    }
+
+    /** {@link PasscodeUnlockActivity} is always exempt. */
+    @Override
+    public boolean isExemptActivity(String activityName) {
+        return UNLOCK_CLASS_NAME.equals(activityName) || super.isExemptActivity(activityName);
+    }
+
+    @Override
+    public void onActivityPaused(Activity activity) {
+        if (!isExemptActivity(activity.getClass().getName())) mLostFocusDate = new Date();
+    }
+
+    @Override
+    public void onActivityResumed(Activity activity) {
+        if (!isExemptActivity(activity.getClass().getName()) && shouldShowUnlockScreen()) {
+            Intent i = new Intent(activity.getApplicationContext(), PasscodeUnlockActivity.class);
+            activity.startActivity(i);
```

Comment:
The only change from the `DefaultAppLock` is the removal of the `FLAG_ACTIVITY_NEW_TASK` flag, and using the activity instead of the applicationContext.